python-challenges/web-scraper/devashmit-day17/scraper.py

```
@@ -0,0 +1,35 @@
+               "tags":[t.get_text(strip=True) for t in q.select("a.tag")]}
+              for q in soup.select("div.quote")]
+    nxt = soup.select_one("li.next > a")
+    return quotes, (BASE + nxt["href"]) if nxt else None
+
+def main():
+    print("\nScraping quotes.toscrape.com...")
+    all_q, page, url = [], 1, BASE
+    while url:
+        print(f"  Page {page}: {url}")
+        q, url = scrape_page(url)
+        all_q.extend(q); print(f"    {len(q)} quotes (total: {len(all_q)})")
```

Ruff (0.15.15)

[error] Multiple statements on one line (semicolon)

(E702)